dbt_coves/tasks/setup/all.py

```diff
@@ -0,0 +1,69 @@
+import os
+from pathlib import Path
+
+from rich.console import Console
+
+from dbt_coves.tasks.base import NonDbtBaseTask
+from dbt_coves.config.config import DbtCovesConfig
+
+from .sshkey import SetupSSHTask
+from .git import SetupGitTask
+from .dbt import SetupDbtTask
+from .vs_code import SetupVscodeTask
+from .sqlfluff import SetupSqlfluffTask
+from .pre_commit import SetupPrecommitTask
+
+console = Console()
+
+
+class SetupAllTask(NonDbtBaseTask):
+    """
+    Task that runs ssh key generation, git repo clone and db connection setup
+    """
+
+    key_column_with = 50
+    value_column_with = 30
+
+    @classmethod
+    def register_parser(cls, sub_parsers, base_subparser):
+        subparser = sub_parsers.add_parser(
+            "all",
+            parents=[base_subparser],
+            help="Sets up SSH keys, git repo, and db connections.",
+        )
+        subparser.add_argument(
+            "--templates",
+            type=str,
+            help="Location of your sqlfluff, ci and pre-commit config files",
+        )
+        subparser.set_defaults(cls=cls, which="all")
+        return subparser
+
+    def run(self) -> int:
+        """
+        Env vars that can be set: USER_FULLNAME, USER_EMAIL, WORKSPACE_PATH, GIT_REPO_URL, DBT_PROFILES_DIR
+        """
+        workspace_path = os.environ.get("WORKSPACE_PATH", Path.cwd())
+
+        SetupSSHTask.run()
+
+        SetupGitTask.run(workspace_path)
+
+        config_folder = DbtCovesConfig.get_config_folder(workspace_path=workspace_path)
+
+        context = SetupDbtTask.get_dbt_profiles_context(config_folder)
+
+        SetupDbtTask.dbt_debug(config_folder)
```

Comment:
It doesn't make sense to run `dbt_debug` before `dbt_init`, the sequence should be:
- ssh
- git
- dbt (profiles, init, debug)
- vscode
- sqlfluff
- precommit